OrbitGl/TimeGraph.cpp

```diff
@@ -189,6 +189,24 @@ void TimeGraph::ZoomTime(float a_ZoomValue, double a_MouseRatio) {
   SetMinMax(minTimeUs, maxTimeUs);
 }
 
+void TimeGraph::VerticalZoom(float zoom_value, double mouse_ratio) {
+  static double increment_ratio = 0.1;
+
+  // In this way, VerticalZoom is reversible
+  double scale = zoom_value > 0 ? 1 + increment_ratio : 1 / (1 + increment_ratio);
+
+  double world_height = m_Canvas->GetWorldHeight();
+  double y_mouse_position = m_Canvas->GetWorldTopLeftY() - mouse_ratio * world_height;
+  double top_distance = m_Canvas->GetWorldTopLeftY() - y_mouse_position;
+
+  // As we are scaling things, every item is going to scale its position.
+  double new_y_mouse_position = y_mouse_position * scale;
+  m_Canvas->SetWorldTopLeftY(new_y_mouse_position + top_distance);
```

Comment:
Something is missing in the vertical zoom behavior, we should never see a big empty gap on top of the scheduler track.  This can happen easily when zooming out.  If we then use the slider bar, everything snaps back to the top.

Reply:
It's actually panning that makes it snap back.

Reply:
I didn't know that. Thanks.

After playing a little bit with the layout variables and the slider bar, I realized that the maximum m_WorldTopLeftY is related with the SliderWidth: max (m_WorldTopLeftY) is exactly 1.5* m_SliderWidth.
Shall I maintain this maximum when zooming?